website/views/user.py

```
@@ -979,44 +1134,1123 @@ def badge_user_list(request, badge_id):
+    if currency not in ALLOWED_CURRENCIES:
+        return False, f"Unsupported currency: {currency}"
+    try:
+        usd_amount_dec = Decimal(str(usd_amount))
+    except (InvalidOperation, TypeError, ValueError):
+        return False, "Invalid usd_amount"
+    if usd_amount_dec <= 0:
+        return False, "usd_amount must be > 0"
+
+    try:
+        pr_num = int(pr_number)
+    except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)